dbt_platform_helper/commands/pipeline.py

```diff
@@ -49,20 +50,44 @@ def generate():
     if codestar_connection_arn is None:
         abort_with_error(f'There is no CodeStar Connection named "{app_name}" to use')
 
+    base_path = Path(".")
+    pipelines_dir = base_path / f"copilot/pipelines"
+
+    _clean_pipeline_config(pipelines_dir)
+
     if not is_terraform_project() and "environments" in pipeline_config:
         _generate_environments_pipeline(
-            app_name, codestar_connection_arn, git_repo, pipeline_config["environments"], templates
+            app_name,
+            codestar_connection_arn,
+            git_repo,
+            pipeline_config["environments"],
+            base_path,
+            pipelines_dir,
+            templates,
         )
 
     if "codebases" in pipeline_config:
         account_id, _ = get_account_details()
 
         for codebase in pipeline_config["codebases"]:
             _generate_codebase_pipeline(
-                account_id, app_name, codestar_connection_arn, git_repo, codebase, templates
+                account_id,
+                app_name,
+                codestar_connection_arn,
+                git_repo,
+                codebase,
+                base_path,
+                pipelines_dir,
+                templates,
             )
 
 
+def _clean_pipeline_config(pipelines_dir):
```

Comment:
If you declared this in the method above, it would actually kind of be private, certainly not callable from elsewhere.

Reply:
Python convention is that a preceding underscore means the method is "private". I'm not a big fan of nested method declarations in general. May as well inline it in this case. Extracting it is for readability.

Reply:
extracting it also makes the method testable directly with fair ease, the `_function()` convention is just a way developers convey intention between each other, as there isn't anything at compiler level preventing this being called.

Reply:
Nothing outside of `generate()` needs it, so it probably shouldn't be accessible outside of `generate()`. I would prefer to inline it that make a "not really private" method.

Reply:
Personally I'm happier with the `_methodname` + extraction version than putting it inside something else, as I do know what this implies in Python. Maybe a comment that reiterates the internal only purpose?